add warning for vault lambda extension cache ttl (#19738)
hashicorp/vault: website/content/docs/platform/aws/lambda-extension.mdx

configuration. Setting the header to `recache` will skip the cache lookup and
return and cache the response from Vault as described previously.

> ~> **Warning!** The cached data lives in-memory of the Vault Lambda Extension
> and will be removed when the Lambda execution environment shuts down.

Reviewer: nit:

```suggestion
~> **Warning!** The Vault Lambda Extension's cache is only in-memory
and will not be persisted when the Lambda execution environment shuts down.
```
